### DISCUSSION TITLE
How to generate movement with LIN - Kuka KRL

### DISCUSSION THREAD

**Discussion by AlbertoAlvares** [visose/Robots#240] [Q&A] [answered]
KRL Kuka programs generated by Robots are creating movement only with PTP command. How do I generate commands with LIN?

I like to generate movement with LIN and not PTP. How to configure to generate only LIN?

&ACCESS RVP
&REL 1
DEF CuboSolido20v2_T_ROB1_000()
$TOOL = meltio
$LOAD = meltio_L
$BASE = DefaultFrame
$APO.CDIS = Zone000
PTP {A1 0,A2 -90,A3 90,A4 0,A5 0,A6 0} C_PTP
$BASE = Frame000
$VEL_AXIS[5] = 3.706
PTP {X -9.5,Y -9.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'}
$OUT[1] = TRUE
BAS(#VEL_PTP, 100)
$VEL_AXIS[3] = 0.487
PTP {X -8.5,Y -8.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'}
$OUT[2] = TRUE
BAS(#VEL_PTP, 100)
$VEL_AXIS[0] = 0
PTP {X -8.5,Y -8.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'}
WAIT SEC Wait000
PTP {X -8.5,Y -8.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'}
$OUT[2] = FALSE
PTP {X -8.5,Y -8.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'}
$OUT[3] = TRUE
BAS(#VEL_PTP, 100)
$VEL_AXIS[3] = 0.68
PTP {X 8.5,Y -8.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'} C_PTP
BAS(#VEL_PTP, 100)
$VEL_AXIS[1] = 0.405
PTP {X 8.5,Y 8.5,Z 0,A -90,B 0,C 180,S'B110',T'B110010'} C_PTP
BAS(#VEL_PTP, 100)
$VEL_AXIS[3] = 0.68
PTP {X -8.5,Y 8.5,Z 0,A -90,B 0,C 180,S'B110',T'B110010'} C_PTP
BAS(#VEL_PTP, 100)
$VEL_AXIS[1] = 0.399
PTP {X -8.5,Y -8.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'}
$OUT[3] = FALSE
BAS(#VEL_PTP, 100)
PTP {X -8.5,Y -8.5,Z 0,A -90,B 0,C 180,S'B110',T'B11011'}
$OUT[4] = TRUE

**Comment by visose** [comment, marked as answer]
Change the Motion input of the Create Target component to Linear.
